### DISCUSSION TITLE
Changing the assigned PRBs to the UE does not reflect on the uplink throughput

### DISCUSSION THREAD

**Discussion by alirezalab134** [srsran/srsRAN_Project#904] [General Help]
Hi, 
I followed [this](https://docs.srsran.com/projects/project/en/latest/tutorials/source/near-rt-ric/source/index.html#o-ran-nearrt-ric-and-xapp) tutorial to setup a one UE xApp scenario. Instead of the "kpm_mon_xapp.py" I ran "simple_rc_xapp.py". The changes in the PRB allocation reflect on the Downlink traffic as you can see, but not on the uplink. I generate Uplink traffic using Iperf. Would you please help me figure out what is the problem?
![Screenshot from 2024-10-28 21-03-30](https://github.com/user-attachments/assets/a8656e55-54f1-4467-a801-d2ae4c67dc9d)

**Comment by pgawlowicz** [comment]
Currently, the PRB limits are applied only to PDSCH.

**Reply by alirezalab134** [reply]
I see. Thanks.
Another question which is related to this topic. When I generate traffic using iPerf with very high bandwidth, I see the output of iPerf matching that bandwidth and sending that amount of data however the bandwidth I see on Grafana is limited to 33Mbit/sec and the transmission continues even after iPerf finishes sending, until it transmits the amount of data that iPerf generated originally. To me it seems like the iPerf is sending the data to a buffer and then the buffer sends it to the channel. If that's the case is there any alternative to iPerf so I can generate the traffic and see the result in real time rather than going through a buffer?

**Reply by h57jafari** [reply]
Hi Alireza, I could run KPM mon successfully, but when I run the simple and simple_RC xApps, DL values drop to zero and UE disconnected from network. did you have same problem, could you share your config files for comparison. I may missing some parameters. Thank you.

**Comment by h57jafari** [comment]
Hi Alireza, I can run KPM mon successfully, but for both of simple_xapp and simple_rc_xp I am having problem, where after control send to change PRB values it make UE disconnect and bit rate in DL drop to zero. did you similar problem, did you modify ant parameter/config beyond what mentioned in the document? thank you.

**Comment by pgawlowicz** [comment]
The E2SM-RC PRB quota control was fixed in the newest srsRAN 5G Project release. Could you check?
Please also update the ORAN-SC-RIC repo.

Best,
Piotr
